Update testing practice (#6)
mids-w203/concept_checks: unit_07/hypothesis_test_practice_activity.Rmd

---
title: "Hypothesis Test Practice Activity"
author: "Alex Hughes"
date: "10/4/2021"
output: bookdown::gitbook
---

# Hypothesis Test Practice Activity

> In this short activity, you're going to write, and execute a short series of hypothesis tests using the `R` estimating framework. 

Reviewer: not sure what the R estimating framework is.  just use R?  I don't think short is accurate at the moment :D 
```suggestion
In this activity, you're going to execute a series of hypothesis tests using R. 
```

The data that you will be using comes from *Tidy Tuesday* and is about the costs of university education for undergraduates. 

This data collection effort is described in detail [on Github](https://github.com/rfordatascience/tidytuesday/blob/master/data/2020/2020-03-10/readme.md#tuition_costcsv), but includes the following tables: 

- `tuition_cost`
- `tuition_income` 
- `historial_tuition` 
- `salary_potential` 
- `diversity_school` 

> We are going to ask as series of questions that can be answered with the constrained set of tests that we have available to use from the course. 

Reviewer: just trying to clarify a bit
```suggestion
We are going to ask a series of questions that can be answered with one of the hypothesis tests presented in the course. 
```

For each of the questions, the data *is* available, but you might have to join a table or two, recode a variable or two, or otherwise do a little bit of data work to get the data ready to run the test. 

> For each test that you conduct, please (a) evaluate the assumptions of the test to see if the data satisfy these assumptions; (b) state the null hypothesis that is being evaluated; (c) state the criteria that would lead you to reject the null hypothesis; (d) conduct and interpret the test; and (e) tell us whether the difference you observe between the groups is an *important* one. 

Reviewer: "state the criteria that would lead you to reject the null hypothesis" is it valid for a student to write "p<.05" for every test? perhaps that's a reason to remove this component and test it more directly another way.

historical_tuition %>% 
  filter(year == '1985-86' & tuition_type == 'All Current')
```

Can you directly select the values that are shown in this table? Specifically, suppose that you are interested in the following question: 

> Has the cost of tuition for public schools changed since 1985? 

In order to accomplish this, start from the query written above, but add additional conditions to `filter` on, and also `select` only the columns that you are interested in. 

Once you have formulated your "query", save the resulting values into a numeric object that is called `public_school_cost_1985`. Note that this will require you to convert the type from a data.frame to a numeric value. 

```{r select only public school tuition}
public_school_cost_1985 <- historical_tuition %>% 
  filter(
    year == '1985-86' & tuition_type == 'All Current' & type == 'Public') %>% 
  select(tuition_cost) %>% 
  as.numeric() # This last line is to store the value outside a data.frame.
```

```{r test result}
## so that you can check your results, the value should be $3571.
test_that(
  'public school tuition is $3571', 
  expect_equal(public_school_cost_1985, 3571)
)
```

> Now then, using data that is available to you, please test whether public school tuition has changed from 1985 to the present. In order to do you, you will have to select the appropriate rows and columns of data, and conduct the appropriate test for this data. 

Reviewer: The big issue I see is that you don't know the 1985 tuition level - you only have an estimate, but your one-sample test below treats this as a true value.

To fix this, you could instead create a single variable to be the difference in tuition from 1985 to the present, then test if it's zero.  Students might not initially recognize this as a place to use a one sample test, but it coudl be good learning for them.

1. What is the appropriate test? 
2. What are the assumptions for this test? 

> 3. Are these assumptions for the test satisfied in this case?

Reviewer: I worry that this phrasing makes it sound like we're really interested in a final yes or no answer. I prefer the way we did it on the lab, something like "evaluate each assumption, based on your background knowledge, data visualizations, and numerical summaries."

```{r}
tuition_cost %>% 
  filter(type == 'Public') %>% 
  select(in_state_tuition) %$%
  t.test(
    x=in_state_tuition, 
    mu=public_school_cost_1985
    )
```

> 1. Are the tuition costs different? 

Reviewer: Is this a gotcha question? if students answer yes or no, I would call that wrong - you just have evidence, you haven't proved the hypothesis. perhaps something like the following?
```suggestion
1. Have you found evidence that the tuition costs are different? 
```

> 2. Is this a big difference or a little difference? What makes you think this? 

Reviewer: importance feels more clear to me.
```suggestion
2. Is this an important difference or an unimportant difference? What makes you think this? 
```

  mutate(diverse_school = urm_percentage >= 0.40)
```

```{r plot school urm distribution}
diverse_school %>% 
  filter(urm_percentage < 1) %>% 
  ggplot() + 
  aes(x = urm_percentage) + 
  geom_histogram()
```

```{r}
tuition_cost %>% 
  left_join(diverse_school) %$%
  t.test(in_state_total ~ diverse_school)
```

# Paired t-test 

Do in-state, or out-of-state students pay more for tuition? To begin, create a scatter plot using the `tuition_cost` data. 

- On the x-axis plot the `in_state_total` tuition
- On the y-axis plot the `out_of_state_total` tuition
- Color the points of the plot by the `type` of school. (If it helps, you may filter out the "other" types of schools). 
- If this plot would be enhanced by showing points on different sets of axes, you might consider using the `fact_grid()` layer to your plot, and set either the `rows` or the `cols` argument of this layer to take in the variable `type` using the call `facet_grid(cols = vars(type))`. 

```{r}
tuition_cost %>% 
  filter(type %in% c('For Profit', 'Private', 'Public')) %>%
  ggplot() +  
  aes(x=in_state_total, y=out_of_state_total, color=type) + 
  geom_point(alpha = 0.3) + 
  facet_grid(cols = vars(type))
```

> Given what you have seen, how would you recommend proceeding with your test? Proceed in the way that you think is most reasonable. State the assumptions for your test, evaluate whether they are satisfied, and conduct the test, describing what you have learned about the statistics, and also what the practical meaning of these statistics are. 

Reviewer: the learning is about the effect or the model parameters or the difference in tuition, not the statistics, right?
